// react-liqpay/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var crypto = _interopDefault(require('crypto'));
var PropTypes = _interopDefault(require('prop-types'));

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

function ownKeys(object, enumerableOnly) {
  var keys = Object.keys(object);

  if (Object.getOwnPropertySymbols) {
    var symbols = Object.getOwnPropertySymbols(object);
    if (enumerableOnly) symbols = symbols.filter(function (sym) {
      return Object.getOwnPropertyDescriptor(object, sym).enumerable;
    });
    keys.push.apply(keys, symbols);
  }

  return keys;
}

function _objectSpread2(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {};

    if (i % 2) {
      ownKeys(Object(source), true).forEach(function (key) {
        _defineProperty(target, key, source[key]);
      });
    } else if (Object.getOwnPropertyDescriptors) {
      Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
      ownKeys(Object(source)).forEach(function (key) {
        Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
      });
    }
  }

  return target;
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};

  var target = _objectWithoutPropertiesLoose(source, excluded);

  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = ".styles_buttonSubmit__3Nole {\r\n  border: none !important;\r\n  display:inline-block !important;\r\n  text-align: center !important;\r\n  padding: 7px 20px !important;\r\n  color: #fff !important; \r\n  font-size:16px !important; \r\n  font-weight: 600 !important; \r\n  font-family:OpenSans, sans-serif; \r\n  cursor: pointer !important; \r\n  border-radius: 2px !important;\r\n  background: rgb(122,183,43) !important;\r\n}\r\n\r\n.styles_buttonSubmit__3Nole img {\r\n  margin-right: 7px !important; \r\n  vertical-align: middle !important;\r\n}\r\n\r\n.styles_buttonSubmit__3Nole span {\r\n  vertical-align: middle !important;\r\n}\r\n\r\n.styles_buttonSubmit__3Nole:hover {\r\n  opacity: 0.5;\r\n}\r\n";
var styleFile = {"buttonSubmit":"styles_buttonSubmit__3Nole"};
styleInject(css_248z);

function utf8_to_b64(str) {
  return window.btoa(unescape(encodeURIComponent(str)));
}

var LiqPayPay = function LiqPayPay(_ref) {
  var publicKey = _ref.publicKey,
      privateKey = _ref.privateKey,
      amount = _ref.amount,
      _ref$currency = _ref.currency,
      currency = _ref$currency === void 0 ? 'UAH' : _ref$currency,
      _ref$description = _ref.description,
      description = _ref$description === void 0 ? 'test' : _ref$description,
      _ref$orderId = _ref.orderId,
      orderId = _ref$orderId === void 0 ? Math.floor(1 + Math.random() * 900000000) : _ref$orderId,
      _ref$title = _ref.title,
      title = _ref$title === void 0 ? 'Payment' : _ref$title,
      style = _ref.style,
      props = _objectWithoutProperties(_ref, ["publicKey", "privateKey", "amount", "currency", "description", "orderId", "title", "style"]);

  var jsonString = _objectSpread2({
    public_key: publicKey,
    version: '3',
    action: 'pay',
    amount: amount,
    currency: currency,
    description: description,
    order_id: orderId
  }, props);

  var data = utf8_to_b64(JSON.stringify(jsonString));
  var signString = privateKey + data + privateKey;
  var sha1 = crypto.createHash('sha1');
  sha1.update(signString);
  var signature = sha1.digest('base64');
  return /*#__PURE__*/React.createElement("form", {
    method: "POST",
    action: "https://www.liqpay.ua/api/3/checkout",
    acceptCharset: "utf-8",
    style: _objectSpread2({}, style)
  }, /*#__PURE__*/React.createElement("input", {
    type: "hidden",
    name: "data",
    value: data
  }), /*#__PURE__*/React.createElement("input", {
    type: "hidden",
    name: "signature",
    value: signature
  }), /*#__PURE__*/React.createElement("button", {
    className: styleFile.buttonSubmit
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://static.liqpay.ua/buttons/logo-small.png",
    name: "btn_text"
  }), /*#__PURE__*/React.createElement("span", null, title, " ", amount, " ", currency)));
};

var LiqPaySubscribe = function LiqPaySubscribe(_ref2) {
  var publicKey = _ref2.publicKey,
      privateKey = _ref2.privateKey,
      amount = _ref2.amount,
      _ref2$subscribePeriod = _ref2.subscribePeriodicity,
      subscribePeriodicity = _ref2$subscribePeriod === void 0 ? 'month' : _ref2$subscribePeriod,
      _ref2$currency = _ref2.currency,
      currency = _ref2$currency === void 0 ? 'UAH' : _ref2$currency,
      _ref2$description = _ref2.description,
      description = _ref2$description === void 0 ? 'test' : _ref2$description,
      _ref2$orderId = _ref2.orderId,
      orderId = _ref2$orderId === void 0 ? Math.floor(1 + Math.random() * 900000000) : _ref2$orderId,
      _ref2$title = _ref2.title,
      title = _ref2$title === void 0 ? 'Subscribe' : _ref2$title,
      style = _ref2.style,
      props = _objectWithoutProperties(_ref2, ["publicKey", "privateKey", "amount", "subscribePeriodicity", "currency", "description", "orderId", "title", "style"]);

  var jsonString = _objectSpread2({
    public_key: publicKey,
    version: '3',
    action: 'subscribe',
    subscribe_date_start: Date.now(),
    subscribe_periodicity: subscribePeriodicity,
    amount: amount,
    currency: currency,
    description: description,
    order_id: orderId
  }, props);

  var data = utf8_to_b64(JSON.stringify(jsonString));
  var signString = privateKey + data + privateKey;
  var sha1 = crypto.createHash('sha1');
  sha1.update(signString);
  var signature = sha1.digest('base64');
  return /*#__PURE__*/React.createElement("form", {
    method: "POST",
    action: "https://www.liqpay.ua/api/3/checkout",
    acceptCharset: "utf-8",
    style: _objectSpread2({}, style)
  }, /*#__PURE__*/React.createElement("input", {
    type: "hidden",
    name: "data",
    value: data
  }), /*#__PURE__*/React.createElement("input", {
    type: "hidden",
    name: "signature",
    value: signature
  }), /*#__PURE__*/React.createElement("button", {
    className: styleFile.buttonSubmit
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://static.liqpay.ua/buttons/logo-small.png",
    name: "btn_text"
  }), /*#__PURE__*/React.createElement("span", null, title, " ", amount, " ", currency)));
};

LiqPayPay.propTypes = {
  publicKey: PropTypes.string.isRequired,
  privateKey: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  orderId: PropTypes.any.isRequired,
  title: PropTypes.string.isRequired,
  style: PropTypes.string
};
LiqPaySubscribe.propTypes = {
  publicKey: PropTypes.string.isRequired,
  privateKey: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired,
  subscribePeriodicity: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  orderId: PropTypes.any.isRequired,
  title: PropTypes.string.isRequired,
  style: PropTypes.string
};

exports.LiqPayPay = LiqPayPay;
exports.LiqPaySubscribe = LiqPaySubscribe;
